pkg/config/setup/config.go

```diff
@@ -1880,21 +1896,6 @@ func LoadCustom(config pkgconfigmodel.Config, origin string, secretResolver opti
 		log.Warnf(warningMsg)
 	}
 
-	// We resolve proxy setting before secrets. This allows setting secrets through DD_PROXY_* env variables
-	LoadProxyFromEnv(config)
-
-	if resolver, ok := secretResolver.Get(); ok {
-		if err := ResolveSecrets(config, resolver, origin); err != nil {
-			return &warnings, err
-		}
-	}
-
-	// Verify 'DD_URL' and 'DD_DD_URL' conflicts
-	if EnvVarAreSetAndNotEqual("DD_DD_URL", "DD_URL") {
-		log.Warnf("'DD_URL' and 'DD_DD_URL' variables are both set in environment. Using 'DD_DD_URL' value")
-	}
-
-	useHostEtc(config)
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏When this code was here, it was not executed in the case of an early return when an error occurred on `config.ReadInConfig()` and if `pkgconfigenv.IsServerless()` was true (in line 1866).
Now it is executed anyway because it have been moved in the parent function (`LoadDatadogCustom`).
For example in the mentionned case `LoadProxyFromEnv(config)` would be called 2 times.
Could it cause some undesired side effects ?

Reply:
Nice catch, I didn't notice the function did not return an error for serverless 🙃

Reply:
It increases the time to run the main function of the serverless agent from [19ms](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/569006133) to [23ms](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/567855315), I'll find a way to avoid that

Reply:
Back to [19~20ms](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/569295398) with my fix [`2b68f22` (#27469)](https://github.com/DataDog/datadog-agent/pull/27469/commits/2b68f22e2cb7cca2ad6ff5df4d1a516320e6cc96). 
The issue was that `LoadProxyFromEnv` was running twice.

Reply:
Funny enough I've warn the serverless team that we might be calling this function for the `serverless-init` binary and maybe is not needed. 

https://github.com/DataDog/datadog-agent/pull/25857#discussion_r1658333485

cc @jcstorms1